fastfold/distributed/comm.py

```diff
@@ -27,31 +27,48 @@ def _reduce(tensor: Tensor) -> Tensor:
     return tensor
 
 
-def _split(tensor: Tensor, dim: int = -1) -> Tensor:
+def _split(tensor: Tensor, dim: int = -1, drop_unused=False) -> Tensor:
     if gpc.get_world_size(ParallelMode.TENSOR) == 1:
         return tensor
 
     split_size = divide(tensor.shape[dim], gpc.get_world_size(ParallelMode.TENSOR))
     tensor_list = torch.split(tensor, split_size, dim=dim)
+    rank = gpc.get_local_rank(ParallelMode.TENSOR)
 
-    output = tensor_list[gpc.get_local_rank(ParallelMode.TENSOR)].contiguous()
+    if not drop_unused:
+        output = tensor_list[rank].contiguous()
+    else:
+        output = tensor_list[rank].contiguous().clone()
 
     return output
 
 
-def _gather(tensor: Tensor, dim: int = -1) -> Tensor:
+def _gather(tensor: Tensor, dim=-1, chunk_size=1) -> Tensor:
```

Comment:
Could you please explain your motivation of adding chunk here?

Reply:
My motivation is to further split the Tensor. In the last_block of Evoformer, it can reduce the occupation of gpu memory during gather.

Reply:
OK.
However, as I know, the memory during `gather` is not the bottleneck, or do you have any evidence showing it is?

Reply:
[test_chunk_gather.zip](https://github.com/hpcaitech/FastFold/files/10764897/test_chunk_gather.zip)
When I infer long sequences, this method can save some gpu memory.